crates/ra_hir/src/adt.rs

```diff
@@ -42,33 +42,20 @@ impl StructData {
 }
 
 impl Enum {
-    pub(crate) fn new(def_id: DefId) -> Self {
-        Enum { def_id }
+    pub(crate) fn new(def_id: DefId, variants: Vec<DefId>) -> Self {
```

Comment:
I think `Enum` should stay a wrapper around just the `DefId`, like `Function`, `Module`, and `Struct`. Can't this be part of `EnumData`?

Especially since the `Enum` object itself isn't memoized by salsa -- it's newly constructed everytime the `DefId` is resolved, and that should be cheap.

Reply:
it's now implemented exactly that way :-)